### DISCUSSION TITLE
CompositeRow inside Menu

### DISCUSSION THREAD

**Discussion by hyanmandian** [ariakit/ariakit#3321] [Questions] [answered]
Is there a way to use CompositeRow inside Menu?

I'm trying to do something like:

```
Dropdown
 - Reset
 - Item 1 (Check)  |  Drag
 - Item 2 (Check)  |  Drag
 - Item 3 (Check)  |  Drag
```

In that case I'm trying to achieve the follow behavior:
- Tab = Focus on dropdown
- Press down = Focus on reset
- Press down = Focus on Item 1 (Check)
- Press right = Focus on Item 1 Drag  

The checkboxes are sortable, so that's why I need the "drag" besides the selectable item

[Example code ](https://codesandbox.io/p/sandbox/ariakit-menu-composite-row-vw6v95?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clr9gx9l800063b6jwcql6vsm%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clr9gx9l800023b6j7sgc5q1q%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clr9gx9l800033b6jtbdd3ta8%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clr9gx9l800053b6j7lk75w3b%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clr9gx9l800023b6j7sgc5q1q%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr9gx9l800013b6jr1u4k7j3%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpublic%252Findex.html%2522%257D%255D%252C%2522id%2522%253A%2522clr9gx9l800023b6j7sgc5q1q%2522%252C%2522activeTabId%2522%253A%2522clr9gx9l800013b6jr1u4k7j3%2522%257D%252C%2522clr9gx9l800053b6j7lk75w3b%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr9gx9l800043b6j9p5xq0ch%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clr9gx9l800053b6j7lk75w3b%2522%252C%2522activeTabId%2522%253A%2522clr9gx9l800043b6j9p5xq0ch%2522%257D%252C%2522clr9gx9l800033b6jtbdd3ta8%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clr9gx9l800033b6jtbdd3ta8%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by diegohaz** [comment, marked as answer]
Yes, you should also wrap the other items in `CompositeRow`. Optionally, you can use [`focusShift`](https://ariakit.org/reference/menu-provider#focusshift) to enable moving up and down to rows with fewer items.

However, I suggest reading this discussion https://github.com/ariakit/ariakit/discussions/3276#discussioncomment-7980716 first. In your situation, if it's just about drag buttons, you could possibly get along with the `menu` role. However, the drag button needs to be a menu item, not a generic composite item. As always, test with real users. Using the menu role and standard interactions without rows (which allows the up and down arrow keys to navigate between all items) is the safest option, as I mentioned in that reply.

**Comment by hyanmandian** [comment]
Awesome! Thanks @diegohaz I read the discussion https://github.com/ariakit/ariakit/discussions/3276#discussioncomment-7980716 and I managed to solve it 🙏
